pkg/auth/providers/azure/clients/ad_graph_client.go

```diff
@@ -228,11 +219,10 @@ func (c azureADGraphClient) userToPrincipal(user graphrbac.User) (v3.Principal,
 }
 
 func (c azureADGraphClient) groupToPrincipal(group graphrbac.ADGroup) (v3.Principal, error) {
```

Comment:
Nit: Any reason why this and (and the corresponding user method, both here and in the new client) are `c Client` rather than `c *Client`?

Reply:
Standard go practice - the receiver is not modified so we don't need this to be a pointer.